tests/api_resources/abstract/test_api_resource.py

```diff
@@ -101,3 +113,65 @@ def test_raise_on_incorrect_id_type(self):
         for obj in [None, 1, 3.14, dict(), list(), set(), tuple(), object()]:
             with pytest.raises(stripe.error.InvalidRequestError):
                 self.MyResource.retrieve(obj)
+
+    @pytest.fixture
+    def resource(self):
+        return self.MyResource.construct_from(
+            {"id": "foo"},
+            key="newkey",
+            stripe_version="2023-01-01",
+            stripe_account="acct_foo",
+        )
+
+    @pytest.fixture
+    def check_request_options(self, http_client_mock):
```

Comment:
It might be simpler to directly assert `http_client_mock.assert_requested(` inside a test.